File: benches/mutations/seeds/kotlin/seed1.kt
fun summarizeScores(scores: List<Int>, threshold: Int): List<String> {
    val summaries = mutableListOf<String>()
    var runningTotal = 0
    for (score in scores) {
        if (score < threshold) {
            continue
        }
        runningTotal += score
        var grade = "poor"
        if (score >= 90) {
            grade = "excellent"
        } else if (score >= 50) {
            grade = "adequate"
        }
        summaries.add(grade)
    }
    if (runningTotal > 250) {
        summaries.add("aggregate")
    }
    return summaries
}

File: benches/mutations/seeds/c/seed1.c
/* seed1.c — kernel-styled score-classification helper (C mutation-bench seed,
 * hand-written, mirrors the Go/Rust/Python "summarize_scores" seed family):
 * fixed-size output buffer + out-param, no dynamic allocation, per kernel idiom. */

static int classify_scores(const int *scores, int count, int threshold,
                            const char **grades, int *total)
{
    int written = 0;
    int sum = 0;

    for (int i = 0; i < count; i++) {
        int score = scores[i];

        if (score < threshold) {
            continue;
        }

        sum += score;

        if (score >= 90) {
            grades[written] = "excellent";
        } else if (score >= 50) {
            grades[written] = "adequate";
        } else {
            grades[written] = "poor";
        }

        written++;
    }

    *total = sum;

    if (sum > 250) {
        grades[written++] = "aggregate";
    }

    return written;
}
